lib/AbstractHandle/Utils/MongoUtil.py

```diff
@@ -58,13 +58,11 @@ def _get_collection(self, mongo_host, mongo_port, mongo_database, mongo_collecti
         return my_collection
```

Comment:
This is still subject to race conditions since you fetch the latest ID, then save the object, then update the counter.

The logic needs to be:
1) increment the counter, returning the new value (which is only known to the current thread)
2) put that in the handle object 
3) save the object

Then there's no chance of collisions.

Reply:
👍